logic/src/commonMain/kotlin/com/wire/kalium/logic/data/publicuser/PublicUserMapper.kt

```diff
@@ -1,33 +1,46 @@
 package com.wire.kalium.logic.data.publicuser
 
 import com.wire.kalium.logic.data.id.IdMapper
-import com.wire.kalium.logic.data.notification.LocalNotificationMessageAuthor
 import com.wire.kalium.logic.data.publicuser.model.OtherUser
 import com.wire.kalium.logic.data.user.AvailabilityStatusMapper
 import com.wire.kalium.logic.data.user.ConnectionState
 import com.wire.kalium.logic.data.user.ConnectionStateMapper
 import com.wire.kalium.logic.data.user.UserAvailabilityStatus
 import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.data.user.type.DomainUserTypeMapper
+import com.wire.kalium.logic.data.user.type.UserType
 import com.wire.kalium.logic.di.MapperProvider
 import com.wire.kalium.network.api.model.getCompleteAssetOrNull
 import com.wire.kalium.network.api.model.getPreviewAssetOrNull
 import com.wire.kalium.network.api.user.details.UserProfileDTO
 import com.wire.kalium.persistence.dao.ConnectionEntity
 import com.wire.kalium.persistence.dao.UserAvailabilityStatusEntity
 import com.wire.kalium.persistence.dao.UserEntity
+import com.wire.kalium.persistence.dao.UserTypeEntity
 
 interface PublicUserMapper {
     fun fromDaoModelToPublicUser(userEntity: UserEntity): OtherUser
-    fun fromUserDetailResponse(userDetailResponse: UserProfileDTO): OtherUser
-    fun fromUserApiToEntity(userDetailResponse: UserProfileDTO, connectionState: ConnectionEntity.State): UserEntity
-    fun fromUserDetailResponses(userDetailResponse: List<UserProfileDTO>): List<OtherUser>
-    fun fromPublicUserToLocalNotificationMessageAuthor(author: OtherUser?): LocalNotificationMessageAuthor
+    fun fromUserDetailResponseWithUsertype(
+        userDetailResponse: UserProfileDTO,
+        // UserProfileDTO has no info about userType, we need to pass it explicitly
+        userType: UserType
+    ): OtherUser
+
+    //TODO:I think we are making too complicated parsers,
+    // maybe a good solution will be fetching self user when we are saving other users to db?
```

Comment:
We can leave this for later.

I guess this goes hand-in-hand with what I was saying.

UseCases are smart, if we need `SelfUser` in order to fetch others, the `UseCase` can pass it.


I feel like a `Repository` refactor is boiling. They're too big!